services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,38 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if(event == NULL) return ERR_CODE_INVALID_ARG;
+  if(thermalMgrQueueHandle == NULL) return ERR_CODE_INVALID_STATE;
+  if(xQueueSend(thermalMgrQueueHandle, event, (TickType_t) 0) != pdPASS) return ERR_CODE_INVALID_QUEUE_MSG;
```

Comment:
only way this does not return pdPASS is if the queue is full so we should use that error code instead. xQueueSend has no way of knowing if the message you pass in is valid or invalid, so `ERR_CODE_INVALID_QUEUE_MSG` is mainly used when you receive from a queue and see that the message is invalid

Reply:
ah yes good point. I have changed it to return `ERR_CODE_QUEUE_FULL`